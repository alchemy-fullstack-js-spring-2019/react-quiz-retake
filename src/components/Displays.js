import React from 'react';
import PropTypes from 'prop-types';
import Display from './Display';

function Displays({ memes }) {
  const memeList = memes.map((meme, i) => {
    return (
      <li key={i}>
        <Display headerText={meme.headerText} footerText={meme.footerText} image={meme.image}/>
      </li>
    );
  });

  return (
    <ul>
      {memeList}
    </ul>
  );
}

Displays.propTypes = {
  memes: PropTypes.array.isRequired
};

export default Displays;
