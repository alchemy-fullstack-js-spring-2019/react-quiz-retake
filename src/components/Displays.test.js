import React from 'react';
import { shallow } from 'enzyme';
import Displays from './Displays';

describe('Displays Component', () => {
  it('displays stuffssssssss', () => {
    const memes = [{ headerText: 'text', footerText: 'text', image: 'text' }];
    const wrapper = shallow(<Displays  memes={memes}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
