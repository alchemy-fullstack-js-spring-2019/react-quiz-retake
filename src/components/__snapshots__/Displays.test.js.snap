// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays Component displays stuffssssssss 1`] = `
<ul>
  <li
    key="0"
  >
    <Display
      footerText="text"
      headerText="text"
      image="text"
    />
  </li>
</ul>
`;
